File: src/Cxx/Utilities/TimerLog.cxx
#include <vtkSmartPointer.h>
#include <vtkTimerLog.h>

int main(int, char *[])
{
  vtkSmartPointer<vtkTimerLog> timerLog = 
      vtkSmartPointer<vtkTimerLog>::New();
  
  std::cout << "Current time: " << timerLog->GetUniversalTime() << std::endl;
  
  timerLog->MarkEvent("opened file");
  
  timerLog->MarkEvent("did other stuff");
  
  std::cout << "Timer log:" << *timerLog << std::endl;
  
  return EXIT_SUCCESS;
}

File: src/Java/Utilities/TimerLog.java
import vtk.vtkNativeLibrary;
import vtk.vtkTimerLog;

public class TimerLog 
{
  //-----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  //-----------------------------------------------------------------
		
  public static void main(String s[]) 
  {
    vtkTimerLog TimerLog = new vtkTimerLog();

    System.out.println("Current Time:" + TimerLog.GetUniversalTime());
    TimerLog.MarkEvent("File Opened");
    TimerLog.MarkEvent("Did Somthing");
    System.out.println("Timer Log:" + TimerLog);	     
  }
}
